### PR TITLE
Controller docs

### PR DESCRIPTION
**Pull request by qwandor**
More docs for `homie-controller`.

### CODE REVIEW

File: homie-controller/src/types.rs [hunk 1]
```diff
@@ -75,11 +75,19 @@ impl Display for State {
...
     Float,
+    /// A [boolean value](https://homieiot.github.io/specification/#boolean).
     Boolean,
+    /// A [UTF-8 encoded string](https://homieiot.github.io/specification/#string).
     String,
+    /// An [enum value](https://homieiot.github.io/specification/#enum) from a set of possible
+    /// values specified by the property format.
     Enum,
+    /// An RGB or HSV [color](https://homieiot.github.io/specification/#color), depending on the
+    /// property format.
```
Comment (alsuren) [inline homie-controller/src/types.rs:90]: (optional) can this link to `Property::format`?

File: homie-controller/src/types.rs [hunk 2]
```diff
@@ -157,9 +165,16 @@ pub struct Property {
 
     /// The format of the property, if any. This should be specified if the datatype is `Enum` or
     /// `Color`, and may be specified if the datatype is `Integer` or `Float`.
+    ///
+    /// This field holds the raw string received from the device; use
```
Comment (alsuren) [inline homie-controller/src/types.rs:169]: ; is distracting
```suggestion
    /// This field holds the raw string received from the device. Use
```
(same below)

Comment (qwandor) [reply homie-controller/src/types.rs:169]: Done.

File: homie-controller/src/types.rs (hunk 1 above)
Comment (qwandor) [reply homie-controller/src/types.rs]: Linked to Property::color_format() instead.

File: homie-controller/src/values.rs [hunk 3]
```diff
@@ -4,19 +4,36 @@ use std::num::ParseIntError;
...
+    /// by the device was something different.
     #[error("Expected value of type {expected} but was {actual}.")]
     WrongDatatype {
+        /// The datatype expected by the method call.
         expected: Datatype,
+        /// The actual datatype of the property, as sent by the device.
         actual: Datatype,
     },
+    /// The format of the property couldn't be parsed or didn't match what was expected by the
+    /// method call.
```
Comment (alsuren) [inline homie-controller/src/values.rs:23]: (for another day): is there any way to split these two cases, and/or show "what was expected" to help with debugging?

**Review by alsuren — APPROVED** [review]

File: homie-controller/src/values.rs (hunk 3 above)
Comment (qwandor) [reply homie-controller/src/values.rs:23]: I originally had two different enum variants, but I'm not convinced it's a useful distinction.

As for "what was expected", there's not necessarily an exact expected value. If you call `value::<ColorRGB>` we expect the string "rgb", but if you call `enum_values` or `range` then we just expect it to follow the right format.

**Pull request merged**